### PR TITLE
New package: dpp-10.0.23_1

### PR DESCRIPTION
**Pull request by OoLunar**
#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-LIBC)

#### About
Libdpp is a C++ lightweight and scalable Discord library. With full support for all of the Discord API, we believe we're ready to be put into the Void Linux repositories for use by our users. We've done our best to comply with the guidelines, however please let me know if there are any improvements I could make to this PR.

#### Notes
Previous PRs (by others): #38601, #37554

### CODE REVIEW

File: srcpkgs/dpp/template [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+homepage="https://dpp.dev"
+distfiles="https://github.com/brainboxdotcc/DPP/archive/refs/tags/v${version}.tar.gz"
+checksum=8f9db61c3586a492ada378235300c509e3bc2fc090cef32de0a8241741038df0
+
+dpp-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
+	short_desc+=" - development files"
+	pkg_install() {
+		# If the path already exists, remove it to prevent conflicts
+		filepath="${DESTDIR}/usr/lib/dpp-${version%.*}"
```
Comment (paper42) [inline srcpkgs/dpp/template:22]: how does this create conflicts?

Comment (OoLunar) [reply srcpkgs/dpp/template:22]: This was on the previous PR's. I was unsure if this was still needed. Am I safe to remove it?

Comment (OoLunar) [reply srcpkgs/dpp/template:22]: It's done here https://github.com/void-linux/void-packages/pull/38601/files#diff-1f1bf3035f39fc9c91e206b7219fd65d0544cdc55d7330fddf5f0e5b0d5e2419R25 though I don't know why. My guess was to remove previous files if the package was being reinstalled.

Comment (paper42) [reply srcpkgs/dpp/template:22]: I don't know why the package would be getting reinstalled in the `masterdir/`, install phase can only run once. Does something actually break when you remove it?

Comment (OoLunar) [reply srcpkgs/dpp/template:22]: It... doesn't appear to. I'll go ahead and remove it then

File: srcpkgs/dpp/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+makedepends="libsodium-devel opus-devel zlib-devel openssl-devel json-c++"
+short_desc="C++ Discord API Library for Bots"
+maintainer="Lunar Starstrum <lunar@forsaken-borders.net>"
+license=Apache-2.0
+homepage="https://dpp.dev"
+distfiles="https://github.com/brainboxdotcc/DPP/archive/refs/tags/v${version}.tar.gz"
+checksum=8f9db61c3586a492ada378235300c509e3bc2fc090cef32de0a8241741038df0
+
+dpp-devel_package() {
+	depends="${makedepends} ${sourcepkg}-${version}_${revision}"
```
Comment (paper42) [inline srcpkgs/dpp/template:18]:
```suggestion
	depends="${makedepends} ${sourcepkg}>=${version}_${revision}"
```

**Pull request closed without merging**